### PR TITLE
Feature/86 open api generated code

### LINKED ISSUE #92: Publish openApi specification in npm
Publish openApi specification in npm, implementation of one usage example(get all users)

#76 

## Estimated time
2h

### PR DESCRIPTION
**Pull request by KseniaBel**
generated API and models

### CODE REVIEW

**Comment by bschne** [comment]
@KseniaBel do you know if there is a way to make this handle authentication as well? I think we'll need to add an auth header to all requests, and there are empty `headerParams` objects in most methods that generate the requests, so I was wondering if you can set it up to add this as a parameter to the signatures. Otherwise I guess we'll just add a function call to the auth helper @nehasingh1491 is building in each method, but I figured maybe there's a way to autogenerate this.

**Comment by KseniaBel** [comment]
@bschne , yes it should be possible. I will give a try if you prefer it this way?

But then the authentication on the backend should be done by using standard SSO (openID connect) compliant library not some proprietary implementation. @nehasingh1491

**Comment by bschne** [comment]
That would be cleaner indeed, but I figure we need to pass the token to the backend on each request anyway, even if we do our own implementation. Or am I missing something?

**Comment by KseniaBel** [comment]
> That would be cleaner indeed, but I figure we need to pass the token to the backend on each request anyway, even if we do our own implementation. Or am I missing something?

True.

**Comment by nehasingh1491** [comment]
Do we really need SSO implementation here? Then it means that if user is
already logged in google applications, then on hitting app url redirect
them to map but our app is not a part of google. And regarding
implementation as per my understanding the purpose is to validate request
on server side by google, and there are n number of ways to implement it.
And I did not implemented any propereitary code, it is a library provided
by Google itself and there are methods provided by Google api which
explicitly verify the tokenid.

**Comment by nehasingh1491** [comment]
Adding authentication token in the header is a good idea we can add
separate task for it's implementation, but as far as I understand we would
still need to pass the token with each request and authenticate it.

**Comment by KseniaBel** [comment]
I am not going to comment on this more. I want to start doing my own tasks, I will leave you to decide and implement this part (since it is related to authentication task). You know how to generate code with the openAPI (I may help with generation frontend code). I guess it will be helpful to show it to everyone than we are more independent.

**Comment by bschne** [comment]
Note @KseniaBel @nehasingh1491 I have potentially found another way re. authentication that should also be nicely doable in OpenAPI and a bit cleaner, so I'll look into this tmrw. before merging this client code.

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "react-google-login": "^5.2.2",
     "react-router-dom": "^5.2.0",
     "react-scripts": "^4.0.2",
+    "sopra-fs21-group-02-dogs-api": "^1.0.0",
```
Comment (bschne) [inline package.json:14]: This is great! But as it stands I don't understand where NPM can find this package to install/reference the API code? Do we need to put it in `/node_modules` manually or tell `npm install` where to find it?

Comment (KseniaBel) [reply package.json:14]: I have published it here https://www.npmjs.com/package/sopra-fs21-group-02-dogs-api 
If you have an NPM account, I can invite you as contributor. 
And no manual work is needed, just npm install

Comment (bschne) [reply package.json:14]: That's even better, I didn't even consider the option that you'd actually publish it on the NPM registry! 👍

**Review by bschne — APPROVED** [review]
All good.

**Pull request merged**